mongoOperator/services/MongoService.py

```diff
@@ -1,21 +1,89 @@
 # Copyright (c) 2018 Ultimaker
 # !/usr/bin/env python
 # -*- coding: utf-8 -*-
+import json
+import logging
+import re
+
+from mongoOperator.helpers.MongoResources import MongoResources
+from mongoOperator.models.V1MongoClusterConfiguration import V1MongoClusterConfiguration
+from mongoOperator.services.KubernetesService import KubernetesService
 
 
 class MongoService:
     """
     Bundled methods for interacting with MongoDB.
     """
+    CONTAINER = "mongodb"
+
+    def __init__(self, kubernetes_service: KubernetesService):
+        self.kubernetes_service = kubernetes_service
+
+    def initializeReplicaSet(self, cluster_object: V1MongoClusterConfiguration) -> None:
+        name = cluster_object.metadata.name
+        namespace = cluster_object.metadata.namespace
+        replicas = cluster_object.spec.mongodb.replicas
+
+        pod_name = "{}-0".format(name)
```

Comment:
Doesn't it need to be initialized in all instances that should be part of the replica set? Or will Mongo handle that when you create a connection string that points to all one of which is the 'master'?

Reply:
> Run rs.initiate() on just one and only one mongod instance for the replica set.
https://docs.mongodb.com/manual/reference/method/rs.initiate/

Reply:
Cewl cewl

Reply:
haha I had to check :laughing: